cli/configuration/v1/ocm_config.go

```diff
@@ -0,0 +1,235 @@
+// If a valid configuration file is found, it attempts to decode it into a `v1.Config` struct.
+// If the file cannot be opened or decoded, an error is returned.
+// One can specify additional paths to search for the configuration file in addition to the default locations.
+//
+// Returns:
+//   - *v1.Config: The parsed configuration file.
+//   - error: An error if no valid configuration file is found or if decoding fails.
+func GetOCMConfig(additional ...string) (*Config, error) {
+	paths, err := GetOCMConfigPaths()
+	paths = append(paths, additional...)
+	if err != nil && len(additional) == 0 {
+		return nil, err
+	}
+	cfgs := make([]*Config, 0, len(paths))
+	for _, path := range paths {
+		cfg, err := GetConfigFromPath(path)
+		if err != nil {
+			slog.Error("ocm config path was skipped due to an error loading it",
+				slog.String("path", path),
+				slog.String("error", err.Error()),
+			)
+			continue
+		}
+		cfgs = append(cfgs, cfg)
+	}
+	return FlatMap(cfgs...)
+}
+
+// GetConfigFromPath reads and decodes the YAML configuration file from the specified path.
+//
+// Parameters:
+//   - path (string): The file path of the configuration file.
+//
+// Returns:
+//   - *v1.Config: The decoded configuration struct.
+//   - error: An error if the file cannot be opened or decoded.
+func GetConfigFromPath(path string) (*Config, error) {
+	file, err := os.Open(path)
+	if err != nil {
+		return nil, err
+	}
+	defer func() {
+		err = errors.Join(err, file.Close())
+	}()
+
+	var instance Config
+	if err := scheme.Decode(file, &instance); err != nil {
+		return nil, err
+	}
+	return &instance, nil
+}
+
+// GetOCMConfigPaths searches for the OCM configuration file in the following locations (in order):
+// 1. The path specified in the OCM_CONFIG_PATH environment variable
+// 2. The XDG_CONFIG_HOME directory (if set), or the default XDG home ($HOME/.config), or the user's home directory
+//   - $XDG_CONFIG_HOME/ocm/config
+//   - $XDG_CONFIG_HOME/.ocmconfig
+//   - $HOME/.config/ocm/config
+//   - $HOME/.config/.ocmconfig
+//   - $HOME/.ocm/config
+//   - $HOME/.ocmconfig
+//
+// 3. The current working directory:
+//   - $PWD/ocm/config
+//   - $PWD/.ocmconfig
+//
+// 4. The directory of the current executable:
+//   - $EXE_DIR/ocm/config
+//   - $EXE_DIR/.ocmconfig
+//
+// Returns:
+//   - string: The path of the discovered configuration file.
+//   - error: An error if no configuration file is found.
+func GetOCMConfigPaths() ([]string, error) {
+	var paths []string
+	if path := getFromEnvironment(); path != "" {
+		paths = append(paths, path)
+	}
+	if path := getFromXDGOrHomeDir(); path != "" {
+		paths = append(paths, path)
+	}
+	if path := getFromWorkingDir(); path != "" {
+		paths = append(paths, path)
+	}
+	if path := getFromExecutableDir(); path != "" {
+		paths = append(paths, path)
+	}
+
+	if len(paths) > 0 {
+		return paths, nil
+	}
+
+	return nil, fmt.Errorf("OCM config not found in any known locations")
```

Comment:
Should we print out the known locations as a support-message?

Reply:
Sure!